python/tests/transform_coords_test.py

```diff
@@ -76,7 +76,7 @@ def split(*, a):
 def test_diamond_graph():
     var = sc.arange(dim='a', start=0, stop=4)
     original = sc.DataArray(data=var, coords={'a': var})
-    graph = {('b', 'c'): split, 'd': bc}
+    graph = {'b': split, 'c': split, 'd': bc}
```

Comment:
I didn't follow the whole discussion around this, but i think this looks a bit weird.

I thought the tuple syntax was much more obvious for someone reading the code to understand what was going on.

I'm guessing the order of `b` and `c` in the graph is important. I think it's ok to have a specific order in a tuple, but in the dict it kind of feels strange.

Reply:
No, the order is not important. @jl-wynen @OwenArnold What do you think about the tuple syntax?

Reply:
I don't see a problem with the tuple syntax apart from potential duplicate keys as you wrote above, Simon. Is this a big issue? I thought you solved it already. And I find the tuple more readable than listing the function multiple times. The latter makes it look like the function can do different things as opposed to producing multiple outputs.

Reply:
Ok, I will revert, and add fixes to catch duplicate keys.